### PR TITLE
Random appearance for random pills

### PR DESCRIPTION
**Pull request by whatston3**
## About the PR

Adds a flag to PillComponent that randomly swaps pill state on init, and swaps strange pills to use these.

Dirty needed to propagate state to client.

## Why / Balance

Ever think it's boring that floor pills are always orange and purple?

## How to test

Spawn strange pills.

## Media

The colours.
image

## Requirements

- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

I have a hunch this might fail some of the init and save tests.

**Changelog**

Trivial, no changelog.

### CODE REVIEW

File: Content.Server/_NF/Chemistry/RandomPillSystem.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public sealed class RandomPillSystem : EntitySystem
+{
+    [Dependency] private IRobustRandom _random = default!;
+
+    public const int MaxPill = 21;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PillComponent, ComponentInit>(OnInit);
```
Comment (MilonPL) [inline Content.Server/_NF/Chemistry/RandomPillSystem.cs:15]: mapinit

File: Content.Server/_NF/Chemistry/RandomPillSystem.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    private void OnInit(EntityUid uid, PillComponent component, ref ComponentInit componentInit)
```
Comment (MilonPL) [inline Content.Server/_NF/Chemistry/RandomPillSystem.cs:18]: `entity<T> ent, ref event _`

File: Content.Server/_NF/Chemistry/RandomPillSystem.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+using Content.Shared.Chemistry.Components;
+using Robust.Shared.Random;
+
+namespace Content.Server._NF.Chemistry;
+
+public sealed class RandomPillSystem : EntitySystem
+{
+    [Dependency] private IRobustRandom _random = default!;
```
Comment (MilonPL) [inline Content.Server/_NF/Chemistry/RandomPillSystem.cs:8]: dependencies are readonly

File: Content.Shared/Chemistry/Components/PillComponent.cs [hunk 4]
```diff
@@ -12,4 +12,10 @@ public sealed partial class PillComponent : Component
     [DataField("pillType")]
     [ViewVariables(VVAccess.ReadWrite)]
     public uint PillType;
+
+    /// <summary>
+    /// Frontier: if true, pill appearance will be randomly generated on init.
+    /// </summary>
+    [DataField("random", serverOnly: true)]
```
Comment (MilonPL) [inline Content.Shared/Chemistry/Components/PillComponent.cs:19]: why the tag

File: Resources/Prototypes/Entities/Objects/Specific/Medical/randompill.yml [hunk 5]
```diff
@@ -86,6 +86,8 @@
   - type: RandomFillSolution
     solution: food
     weightedRandomId: RandomFillStrangePill
+  - type: Pill # Frontier
+    random: true # Frontier
```
Comment (MilonPL) [inline Resources/Prototypes/Entities/Objects/Specific/Medical/randompill.yml:90]: does this really need two comments, just put it on the component and not the data

**Review by MilonPL — CHANGES_REQUESTED** [review]
im going to bite you

File: Content.Shared/Chemistry/Components/PillComponent.cs (hunk 4 above)
Comment (whatston3) [reply Content.Shared/Chemistry/Components/PillComponent.cs:19]: Consistency with the crap around it.

File: Resources/Prototypes/Entities/Objects/Specific/Medical/randompill.yml (hunk 5 above)
Comment (whatston3) [reply Resources/Prototypes/Entities/Objects/Specific/Medical/randompill.yml:90]: Yes.

**Comment by whatston3** [comment]
Review addressed, thanks for the feedback.

repo ban milon he mean

**Review by MilonPL — APPROVED** [review]
good enough

**Review by dvir001 — APPROVED** [review]

**Pull request merged**